### ISSUE TITLE
Corefinement Intersection Error

### ISSUE THREAD

**Issue by JamesYangLim** [CGAL/cgal#2939]
## Issue Details

Hi all,

I was trying to do an intersection boolean operation with corefinement between a big box and a small box as shown below in the image. However the function throws an assertion violation inside intersection_of_coplanar_triangles_3.h. 

The source code below is a very simple repeatable error.  Does anyone know why this assertion happens, yet for other small boxes sharing the same face, there is no assertion?

```
CGAL error: assertion violation!
Expression : ipt_curr.type_1!=ON_FACE
File       : c:\vcpkg\installed\x64-windows\include\cgal\polygon_mesh_processing\internal\corefinement\intersection_of_coplanar_triangles_3.h
Line       : 105
Explanation:
Refer to the bug-reporting instructions at http://www.cgal.org/bug_report.html
unknown file: error: C++ exception with description "CGAL ERROR: assertion violation!
Expr: ipt_curr.type_1!=ON_FACE
File: c:\vcpkg\installed\x64-windows\include\cgal\polygon_mesh_processing\internal\corefinement\intersection_of_coplanar_triangles_3.h
Line: 105" thrown in the test body.
```

![image](https://user-images.githubusercontent.com/30457655/37586778-5c32abea-2b98-11e8-85d5-f6049f7cabae.png)

## Source Code

```
    using Mesh = CGAL::Surface_mesh<typename CGAL::Epeck::Point_3>;

    std::string bigBox = 
        R"(OFF
8 12 0
0 5310 100
90 5310 100
0 5400 100
90 5400 100
0 5310 150
90 5310 150
0 5400 150
90 5400 150
3 0 2 3
3 3 1 0
3 4 5 7
3 7 6 4
3 0 1 5
3 5 4 0
3 3 2 7
3 2 6 7
3 1 3 5
3 3 7 5
3 0 4 2
3 4 6 2
)";

    std::string smallBox =
        R"(OFF
8 12 0
0 5350 105
25 5350 105
0 5375 105
25 5375 105
0 5350 110
25 5350 110
0 5375 110
25 5375 110
3 0 2 3
3 3 1 0
3 4 5 7
3 7 6 4
3 0 1 5
3 5 4 0
3 3 2 7
3 2 6 7
3 1 3 5
3 3 7 5
3 0 4 2
3 4 6 2
)";

    std::istringstream iss(bigBox);
    Mesh bigBoxMesh;
    iss >> bigBoxMesh;

    std::istringstream iss2(smallBox);
    Mesh smallBoxMesh;
    iss2 >> smallBoxMesh;

    CGAL::Polygon_mesh_processing::corefine_and_compute_intersection(smallBoxMesh, bigBoxMesh, smallBoxMesh);
```

## Environment

* Operating system (Windows 64 bits):
* Compiler: C++
* Release or debug mode: Release
* CGAL version: v4.11
* Boost version: 1.66.0

**Comment by jjpepper** [comment]
I have also seen this same assertion fire many times, but found it difficult to produce a simple case.  This is a very nice concise example.

**Comment by sloriot** [comment]
Thanks for the report, #2944 should fix the issue.

@jjpepper as soon as you have 2 meshes that produce an error (no matter how large they are) you can report that to me (provided you already checked that none is self-intersecting)

**Comment by jjpepper** [comment]
Excellent.  Thanks @sloriot, I'll keep that in mind.

**Comment by lrineau** [comment]
Fixed by #2956.
